tests/openassetio-python/CMakeLists.txt

```diff
@@ -171,133 +171,86 @@ openassetio_add_pytest_target(
-openassetio_add_test_fixture_dependencies(
-    openassetio.internal.python-pytest.install-deps
-    openassetio.internal.python-venv.create openassetio.internal.python-pytest.download-deps
-)
-
-# Fixture: install codegen test dependencies.
-openassetio_add_test_fixture_target(openassetio.internal.resources-codegen-pytest.install-deps)
-openassetio_add_test_fixture_dependencies(
-    openassetio.internal.resources-codegen-pytest.install-deps
-    # codegen test deps overlap core Python test deps, so install those
-    # first.
-    openassetio.internal.python-pytest.install-deps
-    openassetio.internal.resources-codegen-pytest.download-deps
-)
+# Optional fixture dependency to create a Python environment in the
+# install tree.
+set(_venv_if_enabled )
+if (OPENASSETIO_ENABLE_PYTHON_TEST_VENV)
+    # Fixture: create Python environment.
+    openassetio_add_test_fixture_target(openassetio-python-venv)
+    set(_venv_if_enabled openassetio-python-venv)
+endif ()
 
-# Fixture: install core Python component.
-openassetio_add_test_fixture_target(openassetio.internal.python.install)
-openassetio_add_test_fixture_dependencies(
-    openassetio.internal.python.install
-    openassetio.internal.python-venv.create
-)
 
 # Fixture: install codegen.
-openassetio_add_test_fixture_target(openassetio.internal.resources-codegen.install)
+openassetio_add_test_fixture_target(openassetio.internal.resources-codegen.install-deps)
 openassetio_add_test_fixture_dependencies(
-    openassetio.internal.resources-codegen.install
-    openassetio.internal.python-venv.create
-    openassetio.internal.resources-codegen.download-deps
+    openassetio.internal.resources-codegen.install-deps
+    ${_venv_if_enabled}
 )
 
 
 #-----------------------------------------------------------------------
 # CTest test targets
 
-
 # Core Python component tests.
 openassetio_add_test_target(openassetio.internal.python-pytest)
 openassetio_add_test_fixture_dependencies(
     openassetio.internal.python-pytest
     openassetio.internal.install
-    openassetio.internal.python.install
-    openassetio.internal.python-pytest.install-deps
+    ${_venv_if_enabled}
 )
 
 # Examples tests.
 openassetio_add_test_target(openassetio.internal.resources-examples-pytest)
 openassetio_add_test_fixture_dependencies(
     openassetio.internal.resources-examples-pytest
     openassetio.internal.install
-    openassetio.internal.python.install
-    openassetio.internal.python-pytest.install-deps
+    ${_venv_if_enabled}
 )
 
 # Code generation tests.
 openassetio_add_test_target(openassetio.internal.resources-codegen-pytest)
 openassetio_add_test_fixture_dependencies(
     openassetio.internal.resources-codegen-pytest
     openassetio.internal.install
-    openassetio.internal.python.install
-    openassetio.internal.resources-codegen.install
-    openassetio.internal.resources-codegen-pytest.install-deps
+    openassetio.internal.resources-codegen.install-deps
```

Comment:
[fix] I think maybe something is awol, that if I build with `-DOPENASSETIO_ENABLE_PYTHON_TEST_VENV=OFF`, then the codegen deps still get installed by ctest:

```
cmake ... -DOPENASSETIO_ENABLE_PYTHON_TEST_VENV=OFF
cmake --build  build
cmake --install build
cmake --build build --target openassetio-python-venv
. build/test-venv/bin/activate
ctest --test-dir build

    Start 1: openassetio.internal.install
1/7 Test #1: openassetio.internal.install ..........................   Passed    4.39 sec
    Start 2: openassetio.internal.core-cpp-test
2/7 Test #2: openassetio.internal.core-cpp-test ....................   Passed    0.14 sec
    Start 3: openassetio.internal.core-c-test
3/7 Test #3: openassetio.internal.core-c-test ......................   Passed    0.15 sec
    Start 4: openassetio.internal.resources-codegen.install-deps
4/7 Test #4: openassetio.internal.resources-codegen.install-deps ...   Passed    1.83 sec
    Start 5: openassetio.internal.python-pytest
5/7 Test #5: openassetio.internal.python-pytest ....................***Failed    0.24 sec
    Start 6: openassetio.internal.resources-examples-pytest
6/7 Test #6: openassetio.internal.resources-examples-pytest ........***Failed    0.24 sec
    Start 7: openassetio.internal.resources-codegen-pytest
7/7 Test #7: openassetio.internal.resources-codegen-pytest .........***Failed    0.24 sec
```

(ignore the fails)

Reply:
Aye, that shouldn't have been happening. Dosen't happen anymore, due to the now-correct addition of `${_venv_if_enabled}` to all post-venv targets. (Sorry, with all the force pushing here it's hard to link this to a specific commit)

Reply:
Yeah this was fixed and squashed into d95c55ba